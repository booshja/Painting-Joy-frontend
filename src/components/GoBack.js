// dependencies
import React from "react";
import styled from "styled-components";
// hooks
import { useHistory } from "react-router";

const StyledBtn = styled.button`
    outline: none;
    background: transparent;
    color: #207070;
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    font-weight: 700;
    cursor: pointer;
    border: none;
    align-self: flex-start;
    margin: 0.5rem 0 0.5rem 7.5%;

    i {
        margin-right: 0.5rem;
    }
`;

const GoBack = ({ to }) => {
    // set up history
    const history = useHistory();

    return (
        <StyledBtn
            onClick={() => {
                history.push(to);
            }}
        >
            <i className="fas fa-long-arrow-alt-left"></i>
            Back
        </StyledBtn>
    );
};

export default GoBack;
