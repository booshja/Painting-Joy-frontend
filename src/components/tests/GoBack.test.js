import { render } from "@testing-library/react";
import GoBack from "./GoBack";

/** Smoke Test */
it("renders without crashing", () => {
    render(<GoBack />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<GoBack />);
    expect(asFragment()).toMatchSnapshot();
});
