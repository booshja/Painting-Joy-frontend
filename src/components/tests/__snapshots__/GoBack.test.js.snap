// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  outline: none;
  background: transparent;
  color: #207070;
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  font-weight: 700;
  cursor: pointer;
  border: none;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin: 0.5rem 0 0.5rem 7.5%;
}

.c0 i {
  margin-right: 0.5rem;
}

<button
    class="c0"
  >
    <i
      class="fas fa-long-arrow-alt-left"
    />
    Back
  </button>
</DocumentFragment>
`;
